apps/search/api/serializers.py

```diff
@@ -27,3 +22,24 @@ def get_meta(self, model) -> Optional[dict]:
             if key in meta:
                 del meta[key]
         return model.meta.to_dict()
+
+
+class DrfSearchableModelSerializer(DrfModelSerializer):
+    """Base serializer for searchable models."""
+
+    meta = serializers.SerializerMethodField(required=False)
+
+    def get_meta(self, model) -> Optional[dict]:
+        if not hasattr(model, 'meta'):
+            return None
+        meta = model.meta.to_dict()
+        for key in ELASTICSEARCH_META_FIELDS_TO_CLEAN:
+            # TODO: find out why in some cases it's already cleaned
+            if key in meta:
+                del meta[key]
+        return model.meta.to_dict()
```

Comment:
Not sure if I'm missing something here, but maybe the logic is broken? Assuming `to_dict()` is just returning a dictionary representation of `model.meta`, is it possible that deleting keys on that dictionary is not mutating `model.meta`? 
Then we just return `to_dict()` again, so maybe nothing has been modified. Do we want to return `meta` instead of calling `to_dict` again?

Reply:
Good catch! First I couldn't understand why this is actually working (i.e cleans given keys). 
After reading some `elasticsearch-dsl` code, it looks like [`to_dict()`](https://github.com/elastic/elasticsearch-dsl-py/blob/7874040b4f1346552a77b6f2a16321f2e1a6c722/elasticsearch_dsl/utils.py#L196-L197) actually returns it's internal `dict` instead returning a new dict. 

So when we're mutating `meta`, we're mutating `model.meta.to_dict()` to because they both reference to same dictionary (tbh, I'm assuming this). This also explains `TODO: find out why in some cases it's already cleaned`

I'm wondering whether we need do to clone the returned dict and return `meta` to make it cleaner.